### PR TITLE
feat/utf8: add test for unicode chars

### LINKED ISSUE #78: thread 'main' panicked at 'byte index 40 is not a char boundary; it is inside '♫'
**Describe the bug**

Not handling UTF-8 well

**To Reproduce**
Steps to reproduce the behaviour:

in rust-lang/rust (52d2c7ac948d6abdc18eb9e05a53d03ddcaffd98)

```
> env RUST_BACKTRACE=1 cargo spellcheck
thread 'main' panicked at 'byte index 40 is not a char boundary; it is inside '♫' (bytes 38..41) of `///         println!("♫ Still alive ♫");`', /home/ivan/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/str/mod.rs:1920:47
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1076
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1537
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:218
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:490
  11: rust_begin_unwind
             at src/libstd/panicking.rs:388
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:101
  13: core::str::slice_error_fail
             at src/libcore/str/mod.rs:0
  14: core::str::traits::<impl core::slice::SliceIndex<str> for core::ops::range::Range<usize>>::index::{{closure}}
  15: cargo_spellcheck::documentation::literalset::LiteralSet::into_chunk
  16: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
  17: cargo_spellcheck::documentation::Documentation::extend
  18: cargo_spellcheck::run
  19: cargo_spellcheck::main
  20: std::rt::lang_start::{{closure}}
  21: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  22: std::panicking::try::do_call
             at src/libstd/panicking.rs:297
  23: std::panicking::try
             at src/libstd/panicking.rs:274
  24: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  25: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  26: main
  27: __libc_start_main
  28: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Expected behavior**

No can do panic.

**Screenshots**
If applicable, add screenshots or copies of the commandline output to help explain your problem.
Use code blocks

**Pplease complete the following information:**
 - System: [e.g. Fedora, Ubuntu, Win10, MacOS] arch linux
 - Obtained: [i.e. git or cargo] cargo
 - Version: `$(cargo spellcheck --version)` 0.3.0-beta.1

**Additional context**

https://github.com/drahnr/cargo-spellcheck/issues/77

### PR DESCRIPTION
**Pull request by drahnr**
## What does this PR accomplish?

Handle UTF8 chars properly 🚤🌴

 * Bug Fix

Closes #78 

## Changes proposed by this PR:

All ranges are characters, all lengths are in chars too.

### CODE REVIEW

**Comment by drahnr** [comment]
@KuabeM I'll merge this and resolve any review comments as separate issues later on

File: src/documentation/literalset.rs [hunk 1]
```diff
@@ -24,7 +24,8 @@ impl LiteralSet {
     ///
     /// Returns literl within the Err variant if not adjacent
     pub fn add_adjacent(&mut self, literal: TrimmedLiteral) -> Result<(), TrimmedLiteral> {
-        let previous_line = literal.span().end.line;
+        dbg!((literal.as_str(), literal.pre(), literal.post()));
```
Comment (KuabeM) [inline src/documentation/literalset.rs:27]: Is this `dbg!` intentional?

**Comment by KuabeM** [comment]
Looks good to me :smile:

File: src/documentation/literalset.rs (hunk 1 above)
Comment (drahnr) [reply src/documentation/literalset.rs:27]: Fixed in master already :>
